src/masternodes/rpc_accounts.cpp

```diff
@@ -1538,69 +1574,77 @@ UniValue accounthistorycount(const JSONRPCRequest& request) {
     LOCK(cs_main);
     CCustomCSView view(*pcustomcsview);
     CCoinsViewCache coins(&::ChainstateActive().CoinsTip());
-
-    CScript lastOwner;
     uint64_t count = 0;
-    auto lastHeight = uint32_t(::ChainActive().Height());
-    const auto currentHeight = lastHeight;
 
-    auto shouldContinueToNextAccountHistory = [&](AccountHistoryKey const & key, AccountHistoryValue value) -> bool {
-        if (!owner.empty() && owner != key.owner) {
-            return false;
-        }
+    for (const auto &owner : accountSet) {
```

Comment:
Each loop with owner then loops over the account history with ForEachAccountHistory. Ideally we only want to loop over the entire account history with ForEachAccountHistory once. Can we update shouldContinueToNextAccountHistory to check if accountSet is not empty and if so check if the current owner is in that set?